### PR TITLE
SRG-APP-000039-CTR-000110: Create appropriate checks and remediations

### PR DESCRIPTION
**Pull request by jhrozek**
#### Description:

- OCP4: Add a new rule `project_config_and_template_network_policy` - adds a rule that checks if a project template exists
  that defines default NetworkPolicies as well as checks if a project template is used in the cluster config
- OCP4 STIG: SRG-APP-000039-CTR-000110 is resolved by `project_config_and_template_network_policy` - makes sure
  the control is covered by the rule added in the previous commit

#### Rationale:

- We didn't have rule coverage for this control

#### Review Hints:

- Check out the rules

- Compare the built SRG mapping with the WIP DISA STIG spreadsheet

### CODE REVIEW

**Comment by jhrozek** [comment]
It is expected that tests will fail until https://github.com/ComplianceAsCode/compliance-operator/pull/207 is merged and released.

**Comment by jhrozek** [comment]
huh, interesting:
```
    helpers.go:808: Result - Name: e2e-stig-project-config-and-template-network-policy - Status: FAIL - Severity: medium
    helpers.go:815: E2E-FAILURE: The expected result for the project_config_and_template_network_policy rule didn't match. Expected 'PASS', Got 'FAIL'
```
the rule did work for me locally. Need to investigate.

**Comment by Vincent056** [comment]
> huh, interesting:
> 
> ```
>     helpers.go:808: Result - Name: e2e-stig-project-config-and-template-network-policy - Status: FAIL - Severity: medium
>     helpers.go:815: E2E-FAILURE: The expected result for the project_config_and_template_network_policy rule didn't match. Expected 'PASS', Got 'FAIL'
> ```
> 
> the rule did work for me locally. Need to investigate.

Does CO have permission for it?

**Comment by jhrozek** [comment]
On Thu, Feb 02, 2023 at 11:54:48PM -0800, Vincent Shen wrote:
> > huh, interesting:
> > 
> > ```
> >     helpers.go:808: Result - Name: e2e-stig-project-config-and-template-network-policy - Status: FAIL - Severity: medium
> >     helpers.go:815: E2E-FAILURE: The expected result for the project_config_and_template_network_policy rule didn't match. Expected 'PASS', Got 'FAIL'
> > ```
> > 
> > the rule did work for me locally. Need to investigate.
> 
> Does CO have permission for it? 

ha, probably not, I forgot after sending the patch :-)

**Comment by jhrozek** [comment]
This should now hopefully pass since we released the new CO version.

**Comment by jhrozek** [comment]
bah, not a single cluster installed
/retest

**Comment by jhrozek** [comment]
Looks like the new CO is not on quay yet...

**Comment by jhrozek** [comment]
I pushed new CO images to quay. Let's try the tests again
/retest

**Comment by jhrozek** [comment]
Ha, I think we're seeing some regressions after I've pushed the new CO version. The rules that are in this PR are finally passing, but some older rules are not.
Going to investigate..

**Comment by jhrozek** [comment]
rebased to pick up CI changes and added newlines

**Comment by jhrozek** [comment]
To review, you can compare the [SRG export artifact](https://github.com/ComplianceAsCode/content/suites/11089079382/artifacts/564480134) with the current [STIG draft](https://docs.google.com/spreadsheets/d/1ouRalvXzRHuhPPN_AKMvqqscUqGvVOkkEdMP3YYBO44/edit#gid=1859516253)

File: applications/openshift/networking/project_config_and_template_network_policy/rule.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    <pre>oc get -nopenshift-config template -ojson |  jq -r '{{{ jqfilter }}}'</pre>
+    If there is at least one matching template, the above query would
+    return true.
+
+    Verify that the project config named <tt>cluster</tt> references the
+    projectRequestTemplate that defines NetworkPolicies:
+    <pre>oc get project.config.openshift.io/cluster -ojsonpath="{.spec.projectRequestTemplate.name}"</pre>
+
+{{% set api_path = '/apis/template.openshift.io/v1/namespaces/openshift-config/templates' %}}
+{{% set jqfilter = '[.items[] | any(.objects[]?; .kind == "NetworkPolicy") ]' %}}
```
Comment (Vincent056) [inline applications/openshift/networking/project_config_and_template_network_policy/rule.yml:56]: I think you might need to move 
```
{{% set api_path = '/apis/template.openshift.io/v1/namespaces/openshift-config/templates' %}}
{{% set jqfilter = '[.items[] | any(.objects[]?; .kind == "NetworkPolicy") ]' %}}
```
before referencing them, it is missing from rendered datastream

Comment (jhrozek) [reply applications/openshift/networking/project_config_and_template_network_policy/rule.yml]: thank you for the careful review, fixed

**Comment by jhrozek** [comment]
The latest [SRG export](https://github.com/ComplianceAsCode/content/suites/11277235344/artifacts/578308334) can be compared with the latest [STIG draft](https://docs.google.com/spreadsheets/d/1ouRalvXzRHuhPPN_AKMvqqscUqGvVOkkEdMP3YYBO44/edit#gid=959715005)

**Review by Vincent056 — APPROVED** [review]
/lgtm

**Pull request merged**